frontend/src/components/smart/PvPArenaSummary.vue

```diff
@@ -144,14 +157,61 @@ export default {
   },
 
   methods: {
+    ...mapActions([
+      'withdrawFromArena',
+      'getDuelQueue'
+    ]),
+
     setTab(tabNumber) {
       this.tab = tabNumber;
     },
 
+    handleErrorMessage(value, errorMessage, returnedMessage) {
+      if (value.includes(`reverted with reason string '${errorMessage}'`)) {
+        return this.$dialog.notify.error(returnedMessage);
+      }
+      return 'There has been an error. Try again.';
+    },
+
     async handleEnterArenaClick() {
       this.$emit('enterMatchMaking');
     },
+
+    async leaveArena() {
+      this.loading = true;
+
+      if (this.isCharacterInDuelQueue) {
+        alert(i18n.t('pvp.currentlyInDuel'));
+        this.loading = false;
+        return;
+      }
+
+      try {
+        await this.withdrawFromArena(this.currentCharacterId);
+
+        this.$emit('leaveArena');
+      } catch (err) {
+        console.log('leave arena error: ', err.message);
+
+        this.handleErrorMessage(err.message, 'Char not in arena', 'The character is not in the arena');
+        this.handleErrorMessage(err.message, 'Defender duel in process', 'Duel already in process');
+      }
+
+      this.loading = false;
```

Comment:
I would place it in `finally` block, just for the sake of keeping it tidy, also, not sure if this gets called on try fail